src/UserSpaceInstrumentation/InjectLibraryInTracee.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_INJECT_LIBRARY_IN_TRACEE_H_
+#define USER_SPACE_INSTRUMENTATION_INJECT_LIBRARY_IN_TRACEE_H_
+
+#include <sys/types.h>
+
+#include <cstdint>
+#include <filesystem>
+#include <string_view>
+
+#include "OrbitBase/Result.h"
+
+namespace orbit_user_space_instrumentation {
+
+// Open, use, and close dynamic library in the tracee. The functions here resemble the respective
+// functions offered by libdl as documented e.g. here: https://linux.die.net/man/3/dlopen. We rely
+// on libc being loaded into the tracee.
```

Comment:
libdl? Also since we currntly are not relying on libdl's dlopen but rather on libc's private '__libc_dlopen_mode@@GLIBC_PRIVATE' symbol, is it guaranteed to work if libdl is not loaded?

Reply:
(answering my own question: __libc_dlopen_mode calls into ld-linux.so and does not need libdl.so)

Reply:
As dicuseed offline ...